app/src/main/java/com/surrus/peopleinspace/di/AppModule.kt

```diff
@@ -1,9 +1,12 @@
 package com.surrus.peopleinspace.di
 
+import com.surrus.common.di.createDbClient
+import com.surrus.common.repository.AndroidDatabaseDependencyProvider
 import com.surrus.peopleinspace.ui.PeopleInSpaceViewModel
 import org.koin.androidx.viewmodel.dsl.viewModel
 import org.koin.dsl.module
 
 val appModule = module {
     viewModel { PeopleInSpaceViewModel(get(),get()) }
+    single { createDbClient(AndroidDatabaseDependencyProvider(get())) }
```

Comment:
Wonder if there's way this can be done without `app` module knowing anything about db client etc.  Would like to keep it abstracted if possible from how persistence etc is managed in the shared module.

Reply:
That makes sense, but I couldn't think of any other way to do this, as the Android counterpart requires context.

https://kotlinlang.org/docs/mobile/configure-sqldelight-for-data-storage.html#ios-driver
The docs mention a similar way. 

As far I see, as long as there are some platform-specific dependencies, we have to inject them at the respective entry points. 

What do you think?

Reply:
I think it's worth looking at approach used in https://github.com/touchlab/KaMPKit (they're also using Koin and SQLDelight)....I had meant to look at adopting approach they were using but hadn't had chance yet.

Reply:
Oh, thanks! This actually makes sense, the context is already injected into the koin container. 

But I didn't quite like this approach, as it's making the assumption that context will be provided via get(). (Shared module is assuming that it will be provided by the platform) 
See [https://github.com/touchlab/KaMPKit/blob/main/shared/src/androidMain/kotlin/co/touchlab/kampkit/KoinAndroid.kt](here)
Is that the right thing to do? (Genuinely asking to know the right approach here)

Reply:
Yeah, I know what you mean....definitely a trade off between the approaches.  What I do like about the KampKit approach though is that things are mostly encapsulated within shared code (other than of course providing the context like it's doing).  If you were to say swap out SQLDelight with some other persistence approach or generally change how the repository is implemented (e.g. to using GraphQL) then you wouldn't need to potentially have to change all the clients.

Reply:
yes, that's true. I get it now. Thank you! 

Shall I revert this and implement it similar to the way they've done? 
it'll only require changes to the expect / actual implementations and not any platform-specific modules.

Reply:
Yeah, if you can make that update that would be great.  Thanks!